src/HMRServer.js

```diff
@@ -8,11 +8,14 @@ class HMRServer {
     });
 
     this.wss.on('connection', ws => {
+      ws.onerror = this.handleSocketError;
       if (this.unresolvedError) {
         ws.send(JSON.stringify(this.unresolvedError));
       }
     });
 
+    this.wss.on('error', this.handleSocketError);
```

Comment:
This is probably not needed as it adds the listener after the server is bound. errors are forwarded from the internal `http.Server` and most of them (all?) are emitted when listening.